pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -291,9 +291,12 @@ func addDeploymentKeyToReplicationController(oldRc *api.ReplicationController, c
 		oldRc.Spec.Template.Labels = map[string]string{}
 	}
 	oldRc.Spec.Template.Labels[deploymentKey] = oldHash
-	if _, err := client.ReplicationControllers(namespace).Update(oldRc); err != nil {
+
+	rc, err := client.ReplicationControllers(namespace).Update(oldRc)
+	if err != nil {
 		return err
 	}
+	oldRc.ResourceVersion = rc.ResourceVersion
```

Comment:
You can't actually do this safely, as you might accidentally over-write fields of the rc on the server by another client since the last GET of oldRc on this client.

You need to take the rc returned by your call to Update above, and use that in place of oldRc from here on forward.